assets/tests/hooks/useVehiclesSocket.test.ts

```diff
@@ -0,0 +1,134 @@
+import { Channel, Socket } from "phoenix"
+import { renderHook } from "react-hooks-testing-library"
+import useVehiclesSocket, {
+  initialState,
+  reducer,
+  removeRoute,
+  setChannelForRoute,
+  setSocket,
+  setVehiclesForRoute,
+} from "../../src/hooks/useVehiclesSocket"
+import { RouteId, Vehicle } from "../../src/skate"
+
+// tslint:disable: react-hooks-nesting
+
+const mockSocketChannel = jest.fn()
+const mockSocketConnect = jest.fn()
+
+jest.mock("phoenix", () => ({
+  Socket: jest.fn(() => ({
+    channel: mockSocketChannel,
+    connect: mockSocketConnect,
+  })),
+  __esModule: true,
+}))
+
+const makeMockChannel = () => {
+  const result = {
+    join: jest.fn(),
+    leave: jest.fn(),
+    on: jest.fn(),
+    receive: jest.fn(),
+  }
+  result.join.mockImplementation(() => result)
+  result.receive.mockImplementation(() => result)
+  return result
+}
+
+describe("useVehiclesSocket", () => {
+  test("vehicles is empty to start with", () => {
+    const { result } = renderHook(() => useVehiclesSocket([]))
+
+    expect(mockSocketConnect).toHaveBeenCalled()
+    expect(result.current).toEqual({})
+  })
+
+  test("selecting a new route subscribes to the new channel", () => {
+    const mockChannel = makeMockChannel()
+    mockSocketChannel.mockImplementationOnce(() => mockChannel)
+
+    const { rerender } = renderHook(() => useVehiclesSocket(["1"]))
+
+    // Needs to be kicked to do the effects again after the socket initializes
+    rerender()
+
+    expect(mockSocketChannel).toHaveBeenCalledTimes(1)
+    expect(mockSocketChannel).toHaveBeenCalledWith("vehicles:1")
+    expect(mockChannel.join).toHaveBeenCalledTimes(1)
```

Comment:
Can we also test the resulting `vehiclesByRouteId` value for each of these tests?

Reply:
I wish. I haven't figured out how to get promises to safely cause effects by resolving in tests yet. That'd make our tests a lot better, so if you think it's worth it I can put more time into figuring that out.

Reply:
I'm happy to trust your judgement and put off perfect tests in favor of moving forward if that seems like the most practical thing for now.

Reply:
Got this to work! By synchronously calling the handlers passed to `.on` and `.receive`, (similar to your synchronously resolving the promise suggestion on the other comment).